### ISSUE TITLE
Maven support?

### ISSUE THREAD

**Issue by mb73** [agido/pageobject#7]
I would like to trigger the tests with Maven. Before switching to PageObject I was using plain ScalaTest together with Maven using the [ScalaTest Maven plugin](http://www.scalatest.org/user_guide/using_the_scalatest_maven_plugin). Using this with PageObject, `mvn test` results in listing the feature and one scenario (out of three) of my test but not really running them (no browser pops up). Output:

> [INFO] --- scalatest-maven-plugin:1.0:test (test) @ integration-testing ---
> Discovery starting.
> [ScalaTest-main] INFO LoginSpec - As a user I want to be able to login (and logout)
> [ScalaTest-main] INFO LoginSpec - As a user I want to be able to login (and logout)
> [ScalaTest-main] INFO LoginSpec - As a user I want to be able to login (and logout)
> [ScalaTest-main] INFO LoginSpec - As a user I want to be able to login (and logout)
> Discovery completed in 226 milliseconds.
> Run starting. Expected test count is: 12
> HtmlUnit:
> Feature: Login page
> Firefox:
> Feature: Login page
> Run completed in 305 milliseconds.
> Total number of tests run: 0
> Suites: completed 2, aborted 0
> Tests: succeeded 0, failed 0, canceled 0, ignored 0, pending 0

Any idea how to get this running?

**Comment by drieks** [comment]
Hi,

I can confirm this. I will have a look.

**Comment by mb73** [comment]
I had a quick look at your [pom](https://github.com/agido/pageobject/blob/master/test/maven/pom.xml). Looks like the paths to the webdrivers are missing, don't they? My plugin config looks like this:

```
    <properties>
        <scala.version>2.11.8</scala.version>
        <scalaBinaryVersion>2.11</scalaBinaryVersion>
        <windowsDrivers>${project.basedir}/src/main/resources/drivers/windows</windowsDrivers>
        <chromeDriver>${windowsDrivers}/chromedriver.exe</chromeDriver>
        <geckoDriver>${windowsDrivers}/geckodriver.exe</geckoDriver>
        <ieDriver>${windowsDrivers}/IEDriverServer.exe</ieDriver>
        <edgeDriver>${windowsDrivers}/MicrosoftWebDriver.exe</edgeDriver>
    </properties>
    <build>
        <sourceDirectory>src/main/scala</sourceDirectory>
        <testSourceDirectory>src/test/scala</testSourceDirectory>
        <plugins>
...
            <plugin>
                <groupId>org.scalatest</groupId>
                <artifactId>scalatest-maven-plugin</artifactId>
                <version>1.0</version>
                <configuration>
                    <reportsDirectory>${project.build.directory}/surefire-reports</reportsDirectory>
                    <junitxml>.</junitxml>
                    <filereports>WDF TestSuite.txt</filereports>
                    <!--
                        argLine MUST BE ON ONE LINE
                    -->
                    <argLine>-Dwebdriver.chrome.driver=${chromeDriver} -Dwebdriver.gecko.driver=${geckoDriver} -Dwebdriver.ie.driver=${ieDriver} -Dwebdriver.edge.driver=${edgeDriver}</argLine>
                </configuration>
                <executions>
                    <execution>
                        <id>test</id>
                        <goals>
                            <goal>test</goal>
                        </goals>
                    </execution>
                </executions>
            </plugin>
        </plugins>
    </build>
```

**Comment by drieks** [comment]
Hi,

yes, I don't added this because it is not needed when using VNC on linux. I added a small howto:
https://github.com/agido/pageobject/blob/develop/howto/maven/pom.xml

(The scalatest plugin in the test pom you mentioned is configured in the parent pom:
https://github.com/agido/pageobject/blob/master/maven/pom.xml#L109-L124)

Test execution using maven is now fixed in 0.2.0-SNAPSHOT. It seems that this was caused by the same error like in https://github.com/agido/pageobject/issues/3 (multi threaded test execution).

Could you please try this snapshot version and tell me if it is working for you? If everything is fine I will release it as 0.2.0.

Add this lines to yours pom.xml file to use the snapshot version:
```xml
    <repositories>
        <repository>
            <id>sonatype-oss-snapshots</id>
            <url>https://oss.sonatype.org/content/repositories/snapshots</url>
            <releases>
                <enabled>false</enabled>
            </releases>
            <snapshots>
                <enabled>true</enabled>
            </snapshots>
        </repository>
    </repositories>
```

**Comment by mb73** [comment]
Looks like I am not able to get the 0.2.0-SNAPSHOT running. With both IDEA and Maven (`mvn -U clean test`) I get the same error:

```
LoginSpec *** ABORTED ***
  java.lang.RuntimeException: Unable to load a Suite class that was discovered in the runpath: LoginSpec
  at org.scalatest.tools.DiscoverySuite$.getSuiteInstance(DiscoverySuite.scala:78)
  at org.scalatest.tools.DiscoverySuite$$anonfun$1.apply(DiscoverySuite.scala:38)
  at org.scalatest.tools.DiscoverySuite$$anonfun$1.apply(DiscoverySuite.scala:37)
  at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:234)
  at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:234)
  at scala.collection.Iterator$class.foreach(Iterator.scala:893)
  at scala.collection.AbstractIterator.foreach(Iterator.scala:1336)
  at scala.collection.IterableLike$class.foreach(IterableLike.scala:72)
  at scala.collection.AbstractIterable.foreach(Iterable.scala:54)
  at scala.collection.TraversableLike$class.map(TraversableLike.scala:234)
  ...
  Cause: java.lang.reflect.InvocationTargetException:
  at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
  at sun.reflect.NativeConstructorAccessorImpl.newInstance(Unknown Source)
  at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(Unknown Source)
  at java.lang.reflect.Constructor.newInstance(Unknown Source)
  at org.scalatest.tools.DiscoverySuite$$anonfun$getSuiteInstance$1.apply(DiscoverySuite.scala:71)
  at org.scalatest.tools.DiscoverySuite$$anonfun$getSuiteInstance$1.apply(DiscoverySuite.scala:64)
  at scala.Option.map(Option.scala:146)
  at org.scalatest.tools.DiscoverySuite$.getSuiteInstance(DiscoverySuite.scala:64)
  at org.scalatest.tools.DiscoverySuite$$anonfun$1.apply(DiscoverySuite.scala:38)
  at org.scalatest.tools.DiscoverySuite$$anonfun$1.apply(DiscoverySuite.scala:37)
  ...
  Cause: java.lang.NoClassDefFoundError: org/pageobject/core/driver/IDriverFactory
  at org.pageobject.scalatest.DriverLaunchWrapper$$anonfun$1$$anonfun$apply$4.apply(DriverLaunchWrapper.scala:90)
  at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:234)
  at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:234)
  at scala.collection.IndexedSeqOptimized$class.foreach(IndexedSeqOptimized.scala:33)
  at scala.collection.mutable.WrappedArray.foreach(WrappedArray.scala:35)
  at scala.collection.TraversableLike$class.map(TraversableLike.scala:234)
  at scala.collection.AbstractTraversable.map(Traversable.scala:104)
  at org.pageobject.scalatest.DriverLaunchWrapper$$anonfun$1.apply(DriverLaunchWrapper.scala:90)
  at org.pageobject.scalatest.DriverLaunchWrapper$$anonfun$1.apply(DriverLaunchWrapper.scala:90)
  at org.pageobject.core.page.UnexpectedPagesFactory$.withMaybeUnexpectedPages(UnexpectedPages.scala:134)
  ...
  Cause: java.lang.ClassNotFoundException: org.pageobject.core.driver.IDriverFactory
  at java.net.URLClassLoader.findClass(Unknown Source)
  at java.lang.ClassLoader.loadClass(Unknown Source)
  at sun.misc.Launcher$AppClassLoader.loadClass(Unknown Source)
  at java.lang.ClassLoader.loadClass(Unknown Source)
  at org.pageobject.scalatest.DriverLaunchWrapper$$anonfun$1$$anonfun$apply$4.apply(DriverLaunchWrapper.scala:90)
  at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:234)
  at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:234)
  at scala.collection.IndexedSeqOptimized$class.foreach(IndexedSeqOptimized.scala:33)
  at scala.collection.mutable.WrappedArray.foreach(WrappedArray.scala:35)
  at scala.collection.TraversableLike$class.map(TraversableLike.scala:234)
  ...
```

I also can’t find `IDriverFactory` in the class path (there is a `InternetExplorerDriverFactory` though).

**Comment by drieks** [comment]
Hi,
I just checked this and downloaded this file:
https://oss.sonatype.org/content/repositories/snapshots/org/pageobject/core_2.11/0.2.0-SNAPSHOT/core_2.11-0.2.0-SNAPSHOT.jar

The class file is included in the jar:
org/pageobject/core/driver/IDriverFactory.class
(also this file: org/pageobject/core/driver/InternetExplorerDriverFactory.class)

Can you try and remove the pageobject snapshot from your local maven repository? Using Linux this is ~/.m2/repository/org/pageobject/core_2.11/0.2.0-SNAPSHOT, on windows maybe "C:\Documents and Settings\{your-username}\.m2". Just removing the directory and rerunning maven should trigger a redownload.

If this does not help, can you please try and checkout the pageobject repository and call "mvn test" in "howto/maven"? This is working fine for me.

**Comment by mb73** [comment]
Very strange. Downloading the file through the browser results in 842 KB, while downloading it through maven always results in 59 KB. 

```
Downloaded: https://oss.sonatype.org/content/repositories/snapshots/org/pageobject/scalatest_2.11/0.2.0-SNAPSHOT/scalatest_2.11-0.2.0-SNAPSHOT.jar (59 KB at 101.5 KB/sec)
```

But wait, your artifactId is `core`. Did that change? I see nothing about `core` in Using [PageObject](https://github.com/agido/pageobject#using-pageobject).

**Comment by drieks** [comment]
core is a transitive dependency of scalatest.

it is possible to add support for another testing framework, maybe junit. all test framework implementations can depend on core where most of pageobject code is located (including IDriverFactory).

scalatest dependent code is located in the scalatest jar, like DriverLauncher and PageObjectSuite.

scalatest jar size is 59K and core jar size is 842K.

**Comment by mb73** [comment]
Sorry, you are right. I deleted the scalatest snapshot but not the core snapshot. Thank you very much!

**Comment by mb73** [comment]
Maybe you could tag this with issue fixed in 0.2.0 or something? Not sure how it works on github.

**Comment by drieks** [comment]
@mb73 pageobject 0.2.0 is now available at maven central
